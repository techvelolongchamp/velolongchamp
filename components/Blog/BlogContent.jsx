import React from 'react'
import { useRouter } from 'next/router'
import PropTypes from 'prop-types'

import { Container, PublishDate } from './BlogContent.styled'

import { formatDate } from '../../utils/date'

const BlogContent = ({ content, date }) => {
  const { locale } = useRouter()
  return (
    <>
      <PublishDate>{formatDate(date, locale)}</PublishDate>
      <Container dangerouslySetInnerHTML={{ __html: content }} />
    </>
  )
}

BlogContent.propTypes = {
  content: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
}

export default BlogContent
