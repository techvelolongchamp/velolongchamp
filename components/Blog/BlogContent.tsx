import React from 'react'
import { useRouter } from 'next/router'
import { TinaMarkdown, TinaMarkdownContent } from 'tinacms/dist/rich-text'
import Image from 'next/image'

import { Container, PublishDate, ImgContainer } from './BlogContent.styled'

import { formatDate } from '../../utils/date'

const components = {
  Button: (props: { href: string; buttonText: string }) => {
    const href = props.href.includes('http') ? props.href : `/${props.href}`
    return (
      <p>
        <a
          className="downloadButton"
          href={href}
          target="_blank"
          rel="noreferrer"
        >
          {props.buttonText}
        </a>
      </p>
    )
  },
  SmallImg: (props: { src: string }) => {
    return (
      <Image
        width="30"
        height="30"
        src={`/media/${props.src}`}
        alt={props.src}
      />
    )
  },
  Img: (props: { src: string; width?: number }) => {
    return (
      <ImgContainer>
        <Image
          src={`/media/${props.src}`}
          alt={props.src}
          width={props.width || 700}
          height={475}
          sizes="100vw"
          style={{
            width: props.width ? `${props.width}px` : '100%',
            height: 'auto',
          }}
        />
      </ImgContainer>
    )
  },
  VideoYoutube: (props: { src: string }) => {
    return (
      <iframe
        width="560"
        height="315"
        src={props.src}
        frameBorder="0"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
      />
    )
  },
}

const BlogContent: React.FC<{
  content: TinaMarkdownContent | TinaMarkdownContent[]
  date: string
}> = ({ content, date }) => {
  const { locale } = useRouter()
  return (
    <>
      <PublishDate>{formatDate(date, locale)}</PublishDate>
      <Container>
        <TinaMarkdown components={components} content={content} />
      </Container>
    </>
  )
}

export default BlogContent
