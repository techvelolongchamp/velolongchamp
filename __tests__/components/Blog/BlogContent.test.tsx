import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import BlogContent from '../../../components/Blog/BlogContent'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('BlogContent unit tests', () => {
  it('should render correctly', () => {
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container } = render(
      <BlogContent
        date="2021-03-24T15:48:12.516Z"
        content="<div>Hello world</div>"
      />
    )

    expect(container).toMatchSnapshot()
  })
})
