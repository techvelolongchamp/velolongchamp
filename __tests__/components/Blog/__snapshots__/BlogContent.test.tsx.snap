// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogContent unit tests should render correctly 1`] = `
.c0 {
  margin: 5px 0;
}

.c1 {
  color: #0d171b;
}

.c1 h2 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

.c1 p {
  font-size: 16px;
  line-height: 26px;
  font-weight: 300;
  margin-bottom: 15px;
}

.c1 hr {
  margin: 15px 0;
  text-shadow: none;
  font-weight: 400;
  border: 0.5px solid;
  color: #999999;
}

.c1 h3 {
  font-size: 18px;
  margin: 20px 0;
  color: #9e9577;
  font-weight: 600;
}

.c1 h4 {
  color: #9e9577;
  font-size: 16px;
  font-weight: 500;
  margin: 15px 0;
  padding-right: 10px;
}

.c1 h5 {
  font-size: 18px;
  margin: 15px 0;
}

.c1 h6 {
  font-size: 16px;
  margin: 15px 0;
  font-weight: 200;
  color: #419a1c;
}

.c1 a {
  color: #419a1c;
  font-weight: 300;
}

.c1 a:hover {
  font-weight: 400;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 a.form {
  font-size: 20px;
  font-weight: 400;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 img {
  text-align: center;
  max-width: 500px;
}

.c1 ul,
.c1 ol {
  list-style: inside;
}

.c1 ul,
.c1 ol li {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #6d6e71;
  font-size: 14px;
  font-weight: 200;
  margin: 10px 0;
  padding-left: 20px;
  max-width: 75%;
}

.c1 .downloadButton {
  background: #6d6e71;
  border-radius: 5px;
  border: none;
  color: #fff;
  font-size: 16px;
  outline: none;
  padding: 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 .article8 {
  padding-left: 30px;
  margin-bottom: 20px;
}

.c1 .article8 p {
  font-style: italic;
  font-size: 12px;
  line-height: initial;
  margin-bottom: 5px;
}

.c1 .article8 ul {
  margin: 5px 0;
  font-style: italic;
  font-size: 12px;
  list-style-type: circle;
}

@media (max-width:640px) {
  .c1 ul,
  .c1 ol li {
    width: 100%;
  }
}

@media screen and (max-width:640px) {
  .c1 .downloadButton {
    font-size: 14px;
  }
}

<div>
  <p
    class="c0"
  >
    24 mars 2021
  </p>
  <div
    class="c1"
  >
    <div>
      Hello world
    </div>
  </div>
</div>
`;
